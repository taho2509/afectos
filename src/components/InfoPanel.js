import React, {Component} from "react";
import {Button, Panel} from "react-bootstrap";
import * as PropTypes from "prop-types";

export class InfoPanel extends Component {
    render() {
        return (
            <Panel>
                <Panel.Body>
                    {this.props.content}
                    {this.props.content ?
                        <Button
                            className="pull-right"
                            bsSize="xsmall"
                            bsStyle={"danger"}
                            onClick={() => this.props.handleClick()}
                        >
                            X
                        </Button>
                        : null
                    }
                </Panel.Body>
            </Panel>);
    }
}

InfoPanel.propTypes = {content: PropTypes.string};