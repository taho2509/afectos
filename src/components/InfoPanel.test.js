import React from 'react';
import {InfoPanel} from './InfoPanel';
import {Button, Panel} from "react-bootstrap";
import Enzyme, {mount, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

//here I setup all tests for InfoPanel
describe("InfoPanel", () => {
    let props;
    let shallowedInfoPanel;
    const infoPanel = () => {
        if (!shallowedInfoPanel) {
            shallowedInfoPanel = shallow(
                <InfoPanel {...props} />
            );
        }
        return shallowedInfoPanel;
    };

    beforeEach(() => {
        props = {
            content: undefined,
        };
        shallowedInfoPanel = undefined;
    });

    it('renders correctly', () => {
        const testString = "Test string";
        beforeEach(() => {
            props.content = testString;
        });
        const tree = infoPanel();
        expect(toJson(tree)).toMatchSnapshot();
    });

    it("always renders a `Panel`", () => {
        expect(infoPanel().find(Panel).length).toBe(1);
    });

    it("always renders a `Panel.Body`", () => {
        expect(infoPanel().find(Panel.Body).length).toBe(1);
    });

    describe("the rendered Panel", () => {
        it("contains everything else that gets rendered", () => {
            const panel = infoPanel().find(Panel);
            const wrappingPanel = panel.first();
            expect(wrappingPanel.children()).toEqual(infoPanel().children());
        });
    });

    describe("when `content` is defined", () => {
        const testString = "Test string";
        beforeEach(() => {
            props.content = testString;
        });

        it("text from `content` is shown", () => {
            const panelBody = infoPanel().find(Panel.Body);
            expect(panelBody.contains(testString)).toBe(true);
        });

        it("Button is shown", () => {
            const panelBody = infoPanel().find(Panel.Body);
            expect(panelBody.find(Button).length).toBe(1);
        });
    });

    describe("when `content` is undefined", () => {
        beforeEach(() => {
            props.content = undefined;
        });

        it("no text is shown", () => {
            const panel = mount(
                <InfoPanel/>
            );
            expect(panel.find(Panel.Body).text()).toEqual("");
        });

        it("Button isn't shown", () => {
            const panelBody = infoPanel().find(Panel.Body);
            expect(panelBody.find(Button).length).toBe(0);
        });
    });
});