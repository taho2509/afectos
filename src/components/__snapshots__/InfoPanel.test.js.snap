// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel renders correctly 1`] = `
<Uncontrolled(Panel)>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  />
</Uncontrolled(Panel)>
`;
